Source/JavaScriptCore/runtime/Structure.cpp

```diff
@@ -340,6 +341,11 @@ Structure::~Structure()
 
     if (isBrandedStructure())
         static_cast<BrandedStructure*>(this)->destruct();
+
+#if ENABLE(WEBASSEMBLY)
+    if (isWebAssemblyGCStructure())
+        static_cast<WebAssemblyGCStructure*>(this)->destruct();
+#endif
```

Comment:
Why don't we put RTT object directly into instance instead of Structure? Extending Structure is **super** high bar. If the other method works, then we should avoid it.

Reply:
This is now done, and I've left some comments explaining how the code changed to make this work.